autoload/lsp/ui/vim.vim

```diff
@@ -421,7 +421,7 @@ function! s:apply_text_edits(uri, text_edits) abort
         let l:end_line = l:text_edit['range']['end']['line'] + 1
         let l:end_character = l:text_edit['range']['end']['character'] " The end position is exclusive so don't add +1
         let l:new_text = l:text_edit['newText']
-        let l:cmd = l:cmd . printf(" | execute 'normal! %dG%d|v%dG%d|c%s'", l:start_line, l:start_character, l:end_line, l:end_character, l:new_text)
+        let l:cmd = l:cmd . printf(" | set paste | execute 'normal! %dG%d|v%dG%d|c%s' | set nopaste", l:start_line, l:start_character, l:end_line, l:end_character, l:new_text)
```

Comment:
Should we save the previous settings and only toggle back if it wasn’t in paste mode already? What if the user was already in the paste mode?

Reply:
`set paste` is always meant to be temporary, so I wouldn't hesitate about resetting it to `no paste` even when user explicitly done so before running command.

In short, if user was in paste mode and ran command, then (s)he should switch back by him/herself.

Reply:
@hauleth Yes. But it should be in a try-finally.

Reply:
@justinmk yeah. I haven't thought about it that way.